File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
def solution(n, computers):

    visited = [False] * (n+1)
    
    def dfs(v):
        visited[v] = True
        
        for i in range(n):
            if not visited[i] and computers[v][i] == 1:
                dfs(i)
    
    cnt = 0
    for i in range(n):
        if not visited[i]:
            dfs(i)
            cnt += 1

    return cnt

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;
class Solution {
    
    static List<Integer>[] graph;
    static boolean[] visited;
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        graph = new ArrayList[n];
        for(int i = 0; i < n; i++) {
            graph[i] = new ArrayList<>();
            for(int j = 0; j < n; j++) {
                if (i != j && computers[i][j] == 1) {
                    graph[i].add(j);
                } 
            }
        }
        
        System.out.println(Arrays.toString(graph));
        visited = new boolean[n];
        for(int i = 0; i < n; i++) {
            if (!visited[i]) {
                visited[i] = true;
                dfs(i);
                answer++;
            }
        }
        
        return answer;
    }
    
    static void dfs(int v) {
        
        for(int next: graph[v]) {
            if (!visited[next]) {
                visited[next] = true;
                dfs(next);
            }
        }
    }
}